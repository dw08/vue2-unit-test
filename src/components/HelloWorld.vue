<template>
  <div class="hello">
    <h1 data-test="title" @click="changeTitle">{{ title }}</h1>
    <div data-test="loginBtn" class="login-btn" @click="toLogin">去登录</div>
  </div>
</template>

<script>
export default {
  name: 'HelloWorld',
  props: {
    title: {
      type: String,
      default: ""
    }
  },
  methods: {
    changeTitle () {
      this.$emit("update:title", '改变标题');
    },
    toLogin () {
      this.$router.push("/login");
    }
  }
}
</script>
