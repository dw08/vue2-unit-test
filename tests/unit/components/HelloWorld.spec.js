import HelloWorld from "@/components/HelloWorld";

import { shallowMount, createLocalVue, mount } from "@vue/test-utils";

import VueRouter from "vue-router";
const localVue = createLocalVue();
localVue.use(VueRouter);
const router = new VueRouter();
describe('HelloWorld component unit test', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(HelloWorld, {
      propsData: {
        title: "测试"
      }
    })
  })
  it('props title test', async () => {
    await wrapper.vm.$nextTick();
    const title = wrapper.find('[data-test="title"]');
    expect(wrapper.vm.title).toBe("测试");
    expect(title.text()).toBe("测试");
  });
  it('method changeTitle test', async () => {
    await wrapper.vm.$nextTick();
    const title = wrapper.find('[data-test="title"]');

    await title.trigger("click")
    expect(wrapper.emitted('update:title')).toBeTruthy();
    // emitted(eventName)[第几次的触发].toEqual(arguments)
    expect(wrapper.emitted('update:title')[0]).toEqual(['改变标题']);
  });
  it('method toLogin test', async () => {
    const wrapper = shallowMount(HelloWorld, {
      localVue,
      router
    });
    const loginBtn = wrapper.find('[data-test="loginBtn"]');
    await loginBtn.trigger("click");
    expect(wrapper.vm.$route.path).toBe("/login");
  });
  it("快照123", async () => {
    const wrapper = mount(HelloWorld);
    await wrapper.setProps({ title: "test" })
    expect(wrapper.html()).toMatchSnapshot();
  })
});