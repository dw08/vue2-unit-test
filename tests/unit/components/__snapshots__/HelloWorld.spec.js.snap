// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld component unit test 快照123 1`] = `
"<div class=\\"hello\\">
  22
  <h1 data-test=\\"title\\">test</h1>
  <div data-test=\\"loginBtn\\" class=\\"login-btn\\">去登录</div>
</div>"
`;
